livedata/src/test/java/io/getstream/chat/android/livedata/controller/QueryChannelsControllerImplTest.kt

```diff
@@ -0,0 +1,91 @@
+package io.getstream.chat.android.livedata.controller
+
+import androidx.arch.core.executor.testing.InstantExecutorExtension
+import com.nhaarman.mockitokotlin2.*
+import io.getstream.chat.android.client.ChatClient
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.livedata.ChatDomainImpl
+import io.getstream.chat.android.livedata.utils.getOrAwaitValue
+import kotlinx.coroutines.Job
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+
+@ExtendWith(InstantExecutorExtension::class)
+internal class QueryChannelsControllerImplTest {
+    @Test
+    fun `when add channel if filter matches should update LiveData from channel to channel controller`() {
+        val channelController = mock<ChannelControllerImpl>()
+        val sut = Fixture()
+            .givenNewChannelController(channelController)
+            .setupChatControllersInstantiation()
+            .get()
+        val newChannel = Channel(cid = "CID", id = "ID")
```

Comment:
I created a [`Mother` class](https://github.com/GetStream/stream-chat-android-livedata/blob/46cb039c3258548f937d169443449f4854a5f182/livedata/src/test/java/io/getstream/chat/android/livedata/Mother.kt#L140) that helps us creating object. Whatever new parameter is added to this entity, will be handled there, so your tests won't break if a new mandatory field is added.
On the other hand, because it use random values, we could detect if any field cause our tests fail (maybe internally we are using some flag/field that has not been considered on our tests).
Those "Mother methods" use the powerfull of Kotlin to provide default values but allowing us to set our needed attributes when we need some custom object

Reply:
You're right. Mother is very convenient and useful pattern in unit tests